src/modules/SdnDiag.NetworkController/public/Set-SdnNetworkController.ps1

```diff
@@ -0,0 +1,139 @@
+                # check to see if the RestIPAddress is already configured, if so, then cross-compare the value currently configured with the new value
+                # if we are just changing from one IP to another, then we can just update the value using Set-NetworkController
+                if ($getNetworkController.RestIPAddress) {
+                    if ($getNetworkController.RestIPAddress -ieq $RestIpAddress) {
+                        "RestIPAddress is already set to {0}. Aborting operation." -f $getNetworkController.RestIPAddress | Trace-Output -Level:Warning
+                        return
+                    }
+                    else {
+                        "RestIPAddress is currently set to {0}. Changing to {1}." -f $getNetworkController.RestIPAddress, $RestIpAddress | Trace-Output -Level:Warning
+                        $confirm = Confirm-UserInput -Message "Do you want to proceed with operation? [Y/N]:"
+                        if ($confirm) {
+                            "Configuring RestIPAddress to {0}. Operation will take several minutes." -f $RestIpAddress | Trace-Output -Level:Verbose
+                            Set-NetworkController @PSBoundParameters
+                        }
+                        else {
+                            "User has opted to abort the operation. Terminating operation" | Trace-Output
+                            return
+                        }
+                    }
+                }
+
+                # if we changing from RestName to RestIPAddress, then we need to remove the RestName property and add the RestIPAddress property
+                # once we remove the RestUrl property, we need to insert a dummy CIDR value to ensure that the Set-NetworkController operation does not fail
+                else {
+                    "RestIPAddress is not configured. Configuring RestIPAddress to {0}. Operation will take several minutes." -f $RestIpAddress | Trace-Output -Level:Warning
+                    $confirm = Confirm-UserInput -Message "Do you want to proceed with operation? [Y/N]:"
+                    if ($confirm) {
+                        $client.PropertyManager.DeletePropertyAsync("fabric:/NetworkController/GlobalConfiguration", "SDNAPI.$version.RestURL")
+                        $client.PropertyManager.PutPropertyAsync("fabric:/NetworkController/GlobalConfiguration", "SDNAPI.$version.RestIPAddress", "10.65.15.117/27")
+
+                        Start-Sleep -Seconds 30 # wait for the property to be deleted
+                        Set-NetworkController @PSBoundParameters
+                    }
+                    else {
+                        "User has opted to abort the operation. Terminating operation" | Trace-Output
+                        return
+                    }
+                }
+            }
+        }
+
+        return (Get-SdnNetworkController)
+    }
+    catch {
+        "{0}`n{1}" -f $_.Exception, $_.ScriptStackTrace | Trace-Output -Level:Error
+    }
```

Comment:
nit: good to add a finally block and call $client.Dispose() if client is not null.

Reply:
@nijosmsft done